File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Prompt user for postive amount of change owed
    int change;
    do
    {
        change = get_int("Change owed: ");
    }
    while (change < 1);

    // Calculate max number of quarters possible to give
    int quarters, dimes, nickels, pennies, coins;
    quarters = change / 25;

    // Recalculate change to remaining change
    change = change - (quarters * 25);

    // Calculate dimes in the remaining change
    dimes = change / 10;
    change = change - (dimes * 10);

    // Calculate nickels in the remaining
    nickels = change / 5;
    change = change - (nickels * 5);

    // Calaculate pennies in the remaing
    pennies = change / 1;

    // Print total number of coins
    coins = quarters + dimes + nickels + pennies;
    printf("%i coins\n", coins);
}

File: pset6/sentimental-cash/cash.py

from cs50 import get_float

# Variables
quarter = 25
dime = 10
nickel = 5
penny = 1
coins = 0

# Prompt user for change owed
while True:
    change = get_float("Change: ")
    if change >= 0:
        break

# Convert to integer
change = change * 100

# Max number of quarters
while change / quarter >= 1:
    change = change - quarter
    coins += 1

# Max number of dimes
while change / dime >= 1:
    change = change - dime
    coins += 1

# Max number of nickels
while change / nickel >= 1:
    change = change - nickel
    coins += 1

# Max number of pennies
while change / penny >= 1:
    change = change - penny
    coins += 1

# Print minimum number of coins
print(f"{coins}")
